packages/astro/src/types/public/config.ts

```diff
@@ -102,7 +110,9 @@ export interface ViteUserConfig extends OriginalViteUserConfig {
  * Astro User Config
  * Docs: https://docs.astro.build/reference/configuration-reference/
  */
-export interface AstroUserConfig {
+export type AstroUserConfig = AstroUserDefineConfig<never>;
+
+export interface AstroUserDefineConfig<TLocales extends Locales = never> {
```

Comment:
Since this file is being re-exported from `types/index.ts`, the types for `AstroUserDefineConfig` will be public anyway, so maybe it's not worth separating this as two types? `defineConfig` with its generic already exposed it anyways.

If we want to completely hide this, we may need to update `types/index.ts` to only export the ones we need. But I'm not sure how to handle `defineConfig`.

Given that it's hard to handle this, I personally think it's easier to expose generic and call it a day. It's unlikely we need to break it again.

Reply:
I think it's fine to export `AstroUserDefineConfig` and mark it as internal. However, I don't think we should make `AstroUserConfig` generic because we're likely to make breaking changes to generics, eg. by adding more

Reply:
Adding more isn't a breaking change though? The generics should all have `=` fallbacks, like the current one, `= never`. And if there's a breaking change there, wouldn't `defineConfig` also inherit the breaking change?

Reply:
Mmmh I guess you're right, adding types isn't breaking. But I think I still prefer to play it safe by having another type where we can do whatever we want. Changing the shape of a generic is not excluded before the next major imo

Reply:
I still don't understand why that would make it safer, it's already being exposed via `defineConfig()` so any breaking changes between it and `AstroUserDefineConfig` will be linked.

But if you prefer to keep the way it is now (with `@internal`), I won't block the PR. It's not a huge deal at the end of the day 👍